Analysis/src/BaseEventAnalyzer.cpp

```diff
@@ -289,20 +289,29 @@ void BaseEventAnalyzer::ProcessDataSource(const SampleDescriptor& sample, const
 
                 if(ana_setup.period == Period::Run2016 || ana_setup.period == Period::Run2017)
                     l1_prefiring_weight = (*event)->l1_prefiring_weight;
-	       
-                uncs_weight_map[UncertaintySource::L1_prefiring][UncertaintyScale::Central] = 
-                        ana_setup.period == Period::Run2016 || ana_setup.period == Period::Run2017 
+
+                uncs_weight_map[UncertaintySource::L1_prefiring][UncertaintyScale::Central] =
+                        ana_setup.period == Period::Run2016 || ana_setup.period == Period::Run2017
                         ? static_cast<float>((*event)->l1_prefiring_weight) : 1;
-                uncs_weight_map[UncertaintySource::L1_prefiring][UncertaintyScale::Up] = 
-                        ana_setup.period == Period::Run2016 || ana_setup.period == Period::Run2017 
+                uncs_weight_map[UncertaintySource::L1_prefiring][UncertaintyScale::Up] =
+                        ana_setup.period == Period::Run2016 || ana_setup.period == Period::Run2017
                         ? static_cast<float>((*event)->l1_prefiring_weight_up) : 1;
                 uncs_weight_map[UncertaintySource::L1_prefiring][UncertaintyScale::Down] =
-                        ana_setup.period == Period::Run2016 || ana_setup.period == Period::Run2017 
+                        ana_setup.period == Period::Run2016 || ana_setup.period == Period::Run2017
                         ? static_cast<float>((*event)->l1_prefiring_weight_down) : 1;
 
                 auto jet_pu_id_weight_provided = eventWeights_HH->GetProviderT<mc_corrections::JetPuIdWeights>(
                         mc_corrections::WeightType::JetPuIdWeights);
-                jet_pu_id_weight = jet_pu_id_weight_provided->Get(*event);
+                jet_pu_id_weight = jet_pu_id_weight_provided->GetWeight(*event);
+
+                for(auto unc_jet_pu_id : {UncertaintySource::PileUpJetId_eff, UncertaintySource::PileUpJetId_mistag}){
+                    for(UncertaintyScale unc_scale_eval : GetAllUncertaintyScales()) {
+                            const UncertaintySource unc_eval = unc_scale_eval == UncertaintyScale::Central ?
+                                                               UncertaintySource::None : unc_jet_pu_id;
+                            uncs_weight_map[unc_jet_pu_id][unc_scale_eval] = static_cast<float>(
+                                jet_pu_id_weight_provided->GetWeight(*event, unc_eval, unc_scale_eval));
+                    }
+                }
```

Comment:
This for should be inside the `if(unc_source == UncertaintySource::None)`, which is defined below.